### PR TITLE
Supplier.development

### PR DESCRIPTION
**Pull request by Dharmiik-Chauhan**

### CODE REVIEW

File: app/controllers/suppliers_controller.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+  def show
+  end
+
+  def edit
+  end
+
+  def create
+    @supplier = Supplier.new(supplier_params)
+
```
Comment (RaviMariya-MTAJ) [inline app/controllers/suppliers_controller.rb]: remove blank lines

File: app/controllers/suppliers_controller.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    respond_to do |format|
+      if @supplier.save
+        format.html { redirect_to @supplier, notice: 'suppiler was successfully created.' }
+        # format.json { render :show, status: :created, location: @stocks }
```
Comment (RaviMariya-MTAJ) [inline app/controllers/suppliers_controller.rb]: remove commented codes

File: app/controllers/suppliers_controller.rb [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+  end
+
+  def update
+    respond_to do |format|
+      if @supplier.update(supplier_params)
+        format.html { redirect_to @supplier, notice: 'supplier was successfully updated.' }
+        # format.json { render :show, status: :ok, location: @stocks }
+      else
+        format.html { render :edit }
+        # format.json { render json: @stock.errors, status: :unprocessable_entity }
```
Comment (RaviMariya-MTAJ) [inline app/controllers/suppliers_controller.rb]: same here

File: app/views/suppliers/_form.html.erb [hunk 4]
```diff
@@ -0,0 +1,67 @@
+<%= form_with(model: supplier, local: true) do |form| %>
+  <% if supplier.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(suppiler.errors.count, "error") %> prohibited this suppiler from being saved:</h2>
+
```
Comment (RaviMariya-MTAJ) [inline app/views/suppliers/_form.html.erb]: same here

File: app/views/suppliers/_form.html.erb [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+        <li><%= message %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
+<table border="1" align="center" cellspasing="5" cellpadding="5">
+  <tr>
+  <div class="field">
+    <td><%= form.label :supplier_name %></td>
+    <td><%= form.text_field :supplier_name, id: :supplier_name %></td>
```
Comment (RaviMariya-MTAJ) [inline app/views/suppliers/_form.html.erb]: why you have given id in fields?

File: app/views/suppliers/_form.html.erb [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+  </div>
+</tr>
+
+<tr>
+  <div class="field">
+    <td><%= form.label :product_name %></td>
+    <td><%= form.text_field :product_name, id: :supplier_product_name %></td>
```
Comment (RaviMariya-MTAJ) [inline app/views/suppliers/_form.html.erb]: please tell me where are you using this id?

File: app/views/suppliers/_form.html.erb [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+  <div class="field">
+    <td><%= form.label :contact_no %></td>
+    <td><%= form.text_field :contact_no, id: :supplier_contact_no %></td>
+  </div>
+</tr>
+
+<tr>
+  <div class="field">
+    <td><%= form.label :gst_no %></td>
+    <td><%= form.text_field :gst_no, id: :supplier_gst_no %></td>
```
Comment (RaviMariya-MTAJ) [inline app/views/suppliers/_form.html.erb]: same in each field

File: app/views/suppliers/index.html.erb [hunk 8]
```diff
@@ -0,0 +1,37 @@
+<p id="notice"><%= notice %></p>
+
+<h1 align="center">Suppiler</h1>
+
+<table border="1" align="center" cellspasing="5" cellpadding="5">
+  <thead>
+    <tr>
+      <th>supplier_name</th>
```
Comment (RaviMariya-MTAJ) [inline app/views/suppliers/index.html.erb]: why are you using table heading name with underscore? are you copying table field name directly here?

File: app/views/suppliers/show.html.erb [hunk 9]
```diff
@@ -0,0 +1,42 @@
+<p id="notice"><%= notice %></p>
+
+<table border="3" align="center" cellspasing="5" cellpadding="10">
+  <tr>
+    <th> <strong>supplier_name:</strong></th>
```
Comment (RaviMariya-MTAJ) [inline app/views/suppliers/show.html.erb]: fix this table column heading everywhere

File: app/views/suppliers/new.html.erb [hunk 10]
```diff
@@ -0,0 +1,5 @@
+<h1 align="center">New supplier</h1>
+
+<%= render 'form', supplier: @supplier %>
```
Comment (RaviMariya-MTAJ) [inline app/views/suppliers/new.html.erb:3]: do we need to pass `@supplier` here?

**Review by RaviMariya-MTAJ — CHANGES_REQUESTED** [review]
please fix this comments

File: app/controllers/suppliers_controller.rb (hunk 1 above)
Comment (Dharmiik-Chauhan) [reply app/controllers/suppliers_controller.rb]: ok sir

**Pull request merged**